crates/autopilot/src/run_loop.rs

```diff
@@ -169,6 +178,37 @@ impl RunLoop {
                     }
                 }
             }
+            let competition_table = SolverCompetitionDB {
+                competition_simulation_block,
+                auction: CompetitionAuction {
+                    orders: auction
+                        .orders
+                        .iter()
+                        .map(|order| order.metadata.uid)
+                        .collect(),
+                    prices: auction.prices.clone(),
+                },
+                solutions: solutions
+                    .iter()
+                    .map(|(index, response)| SolverSettlement {
+                        solver_address: response.submission_address,
+                        score: Some(Score::Solver(response.score)),
+                        ranking: Some(solutions.len() - index),
+                        orders: response
+                            .orders
+                            .iter()
+                            .map(|o| Order {
+                                id: *o,
+                                ..Default::default()
```

Comment:
It probably makes sense to leave a TODO or open an issue about later removing the fields which can not be populated with colocated drivers.

Also I'm sorry but I my brain must have been AFK when I reviewed https://github.com/cowprotocol/services/pull/1731 because I now realize that we should not expect call data from the `/solve` request. Otherwise solvers will not be able to properly utilize RFQ based liquidity.
I see that you implemented the calldata collection via `/solve` to unblock the colocation efforts but we definitely have to revisit that implementation before releasing the driver.

Reply:
I agree. However, it seems that we don't have to decide until we push external solvers to run their own driver. By that time, we should have `/reveal` endpoint added where the `calldata` will be revealed by the winning driver. 
Do you think I should do it asap? (IMO it does not block us from enabling internal colocation).

Reply:
I agree that it only has to happen before external parties are expected to run their own driver.
It's fine if that only happens after we iron out all the remaining kinks of the new driver.

Reply:
Cool. Added https://github.com/cowprotocol/services/issues/1744 to capture the work around this.

Reply:
Experimented quickly with adding /reveal and ended with https://github.com/cowprotocol/services/pull/1745